### PR TITLE
actionified and formatted - needs doxygenation

### PR DESCRIPTION
**Pull request by siddacious**
This PR adds the basic stuff for GitHub actions and a start to addressing the formatting and documentation requirements.

TODO:
~~The documentation needs to be converted to a Doxygen compatible format
I'm guessing the build will fail because LittlevGL is required to compile the examples and it's not available though the Arduino library manager. We'll have to figure out how we want to handle this~~
 * Examples may need more attention

You'll see there is a rather large amount of dependencies listed now; by default the actions workflow will attempt to compile each example on a number of platforms specified here:
https://github.com/adafruit/Adafruit_LvGL_Glue/blob/actionci/.github/workflows/githubci.yml#L23
which refers to this file:
https://github.com/adafruit/ci-arduino/blob/master/build_platform.py#L66

There is a mechanism blacklist or whitelist an example for a given platform should that help.

Let me now if you have any questions

### CODE REVIEW

**Comment by PaintYourDragon** [comment]
I think LittlevGL is in the Library manager as lv_arduino

File: .github/workflows/githubci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    - uses: actions/checkout@v2
+      with:
+         repository: adafruit/ci-arduino
+         path: ci
+
+    - name: pre-install
+      run: bash ci/actions_install.sh
+
+    - name: test platforms
+      run: python3 ci/build_platform.py main_platforms
```
Comment (siddacious) [inline .github/workflows/githubci.yml]: This line will need to be updated to change the platforms that are built, drawing from here:
https://github.com/adafruit/ci-arduino/blob/master/build_platform.py#L28

**Comment by siddacious** [comment]
@PaintYourDragon Please review the docs:
https://adafruit.github.io/Adafruit_LvGL_Glue/html/class_adafruit___lv_g_l___glue.html

**Comment by siddacious** [comment]
@PaintYourDragon I originally had it building the `hello_gizmo` and `widgets_gizmo` examples for the CPX as well but it looks like it ran out of memory:
https://github.com/adafruit/Adafruit_LvGL_Glue/pull/1/checks?check_run_id=577457962#step:6:90

I thought I heard @ladyada mention having a simple UNO example as well so I originally included it as well and had it only build the featherwing examples but it had the same issue obvs.

**Pull request merged**